File: 0075-sort-colors/0075-sort-colors.java
//Space complexity: O(1)
//time Complexity: O(n)
class Solution {
    public void sortColors(int[] nums) {
        //index to follow through final array
        //use counter to count freqency of each colour
        //set colours accordingly
        int index=0;
        int redCount=0, whiteCount=0, blueCount=0;

        for(int i=0; i<nums.length; i++){
            if(nums[i]==0)
                redCount++;
            else if(nums[i]==1)
                whiteCount++;
            else
                blueCount++;
        }

        while(redCount!=0){
            nums[index]=0;
            redCount--;
            index++;
        }
        while(whiteCount!=0){
            nums[index]=1;
            whiteCount--;
            index++;
        }
        while(blueCount!=0){
            nums[index]=2;
            blueCount--;
            index++;
        }
    }
}

File: 0075-sort-colors/0075-sort-colors.swift
class Solution {
    func sortColors(_ nums: inout [Int]) {
        var t = 0
        var redCount = 0
        var whiteCount = 0
        var blueCount = 0

        for i in 0..<nums.count {
            if nums[i] == 0 {
                redCount += 1
            }
            else if nums[i] == 1 {
                whiteCount += 1
            }
            else {
                blueCount += 1
            }
        }

        while redCount != 0 {
            nums[t] = 0
            t += 1
            redCount -= 1
        }
        while whiteCount != 0 {
            nums[t] = 1
            t += 1
            whiteCount -= 1
        }

        while blueCount != 0 {
            nums[t] = 2
            t += 1
            blueCount -= 1
        }
    }
}